save-plugins/warn-plugin/src/commonMain/kotlin/org/cqfn/save/plugin/warn/utils/Warning.kt

```diff
@@ -108,16 +110,22 @@ internal fun String.extractWarning(warningRegex: Regex,
     "SwallowedException")
 internal fun String.getLineNumber(warningRegex: Regex,
                                   lineGroupIdx: Int?,
+                                  placeholder: String,
+                                  lineNum: Int?,
 ): Int? {
     val groups = warningRegex.find(this)?.groups ?: return null
 
-    val line = lineGroupIdx?.let {
-        try {
-            groups[lineGroupIdx]!!.value.toInt()
-        } catch (e: Exception) {
-            throw ResourceFormatException("Could not extract line number from line [$this], cause: ${e.message}")
+    return lineGroupIdx?.let {
+        groups[lineGroupIdx]!!.value.toIntOrNull() ?: run {
+            val warnLine = groups[lineGroupIdx]!!.value
+            if (warnLine[0] != placeholder[0]) {
+                throw ResourceFormatException("The placeholder in warning must start with a ${placeholder[0]}")
+            }
+            try {
+                warnLine.substringAfterLast(placeholder).toInt() + (lineNum ?: 0) + 1
+            } catch (e: Exception) {
+                throw ResourceFormatException("Could not extract line number from line [$this], cause: ${e.message}")
```

Comment:
```suggestion
                throw ResourceFormatException("Could not extract line number from line [$this], cause: ${e.describe()}")
```
In this method we also output class of exception, which should be useful

Reply:
Fixed